### PR TITLE
Feature/handle token in api

### LINKED ISSUE #57: Handle token in API

### PR DESCRIPTION
**Pull request by laska-sl**

### CODE REVIEW

File: ArcadiaParties.API/Startup.cs [hunk 1]
```diff
@@ -38,6 +39,14 @@ public void ConfigureServices(IServiceCollection services)
             // Temporary use SQLite connection until set connection to Arcadia API
             services.AddDbContext<DataContext>(x => x.UseSqlite(Configuration.GetConnectionString("DefaultConnection")));
 
+            services
+                .AddAuthentication(JwtBearerDefaults.AuthenticationScheme)
+                .AddJwtBearer(options =>
+                {
+                    options.Audience = "a2ccb221-60e2-47b8-b28c-bf88a59f7f4a";
```
Comment (sponkrashin) [inline ArcadiaParties.API/Startup.cs]: All authentication parameters should be moved to application settings.

**Review by sponkrashin — CHANGES_REQUESTED** [review]

File: ArcadiaParties.API/Startup.cs (hunk 1 above)
Comment (laska-sl) [reply ArcadiaParties.API/Startup.cs]: Done.

File: ArcadiaParties.API/Startup.cs [hunk 2]
```diff
@@ -62,8 +65,8 @@ public void ConfigureServices(IServiceCollection services)
                     {
                         Implicit = new OpenApiOAuthFlow
                         {
-                            AuthorizationUrl = new Uri("https://login.microsoftonline.com/fa4e9c1f-6222-443d-a083-28f80c1ffefc/oauth2/authorize"),
-                            TokenUrl = new Uri("https://login.microsoftonline.com/fa4e9c1f-6222-443d-a083-28f80c1ffefc/oauth2/token"),
+                            AuthorizationUrl = new Uri("https://login.microsoftonline.com/"+ OAuthSettings.Tenant + "/oauth2/authorize"),
```
Comment (sponkrashin) [inline ArcadiaParties.API/Startup.cs]: Haven't you heard about string interpolation?

File: ArcadiaParties.API/Startup.cs [hunk 3]
```diff
@@ -43,8 +46,8 @@ public void ConfigureServices(IServiceCollection services)
                 .AddAuthentication(JwtBearerDefaults.AuthenticationScheme)
                 .AddJwtBearer(options =>
                 {
-                    options.Audience = "a2ccb221-60e2-47b8-b28c-bf88a59f7f4a";
-                    options.MetadataAddress = "https://login.microsoftonline.com/fa4e9c1f-6222-443d-a083-28f80c1ffefc/.well-known/openid-configuration";
+                    options.Audience = OAuthSettings.ClientId;
+                    options.MetadataAddress = "https://login.microsoftonline.com/" + OAuthSettings.Tenant + "/.well-known/openid-configuration";
```
Comment (sponkrashin) [inline ArcadiaParties.API/Startup.cs]: I'd make it another way - keep urls themselves in the configuration.

File: ArcadiaParties.API/Startup.cs [hunk 4]
```diff
@@ -25,8 +25,11 @@ public class Startup
         public Startup(IConfiguration configuration)
         {
             Configuration = configuration;
+            OAuthSettings = Configuration.GetSection("OAuth").Get<OAuthSettings>();
         }
 
+        public OAuthSettings OAuthSettings { get; set; }
```
Comment (sponkrashin) [inline ArcadiaParties.API/Startup.cs]: Set is not needed.

**Review by sponkrashin — CHANGES_REQUESTED** [review]

File: ArcadiaParties.API/Startup.cs (hunk 2 above)
Comment (laska-sl) [reply ArcadiaParties.API/Startup.cs]: Sure, but I used to use concatenation. Of course, interpolation is more convenient.

File: ArcadiaParties.API/Startup.cs (hunk 3 above)
Comment (laska-sl) [reply ArcadiaParties.API/Startup.cs]: I've moved whole urls to secret

File: ArcadiaParties.API/Startup.cs (hunk 4 above)
Comment (laska-sl) [reply ArcadiaParties.API/Startup.cs]: Set has been removed

File: ArcadiaParties.API/Startup.cs (hunk 2 above)
Comment (sponkrashin) [reply ArcadiaParties.API/Startup.cs]: Concatenation looks ugly and unnaturally, imho.

**Review by sponkrashin — APPROVED** [review]

**Pull request merged**